Copy edit CLI docs

biosafetylvl5 opened the pull request:
I copy-edited the RST of the CLI docs for grammar, rst syntax, brevity and clarity. No changes to functionality were made.

I moved the automatically generated CLI docs to a new file to tidy up the docs.

jsolbrig commented on docs/source/userguide/command_line.rst:
```diff
@@ -1,508 +1,520 @@
+^^^^^^^^^^^^^^^
 
-:ref:`geoips describe family <geoips_describe_family>`
+``describe <interface_name> family <family_name>`` (or ``fam``) is a
+sub-command of ``describe``
```
Maybe for consistency with ``describe interface`` this should be:

> ``describe <interface_name> family <family_name>`` (or ``fam``) retrieves information about a family.

jsolbrig commented on docs/source/userguide/command_line.rst:
```diff
@@ -1,508 +1,520 @@
+^^^^^^^^^^^^^^^^
 
-:ref:`geoips describe package <geoips_describe_package>`
+``describe package`` is a sub-command of describe.
+It returns the following information about a Package:
```
Maybe for consistency with ``describe interface`` this should be:

> ``describe package`` (or ``fam``) retrieves information about a registered plugin package.

jsolbrig commented on docs/source/userguide/command_line.rst:
```diff
@@ -1,508 +1,520 @@
-    * Relative Path
-    * Signature (if applicable)
-    * Source Names (if applicable)
+``describe plugin`` is a sub-command of describe.
+It returns the following information about a Plugin:
```
Maybe for consistency with ``describe interface`` this should be:

> ``describe plugin`` retrieves information about a specific plugin.

jsolbrig commented on docs/source/userguide/command_line.rst:
```diff
@@ -1,508 +1,520 @@
 
-The GeoIPS Command Line Interface (CLI) is currently under development, but we'll keep
-updated instructions in this documentation that reflect the current state of the CLI.
+The CLI interacts with GeoIPS artifacts.
+A GeoIPS artifact is a piece of data or functionality,
+such as an Interface, Package, Plugin,
+Family, Test Script, Test Dataset, etc.
 
```
A different thought here:

> Interaction with and introspection of GeoIPS is performed using the CLI. The CLI can be used to run, configure, and test GeoIPS. It can also print lists of available artifacts and descriptions of individual artifacts.

biosafetylvl5 replied:
This is a bit clunky: "Interaction with and introspection of GeoIPS is performed using the CLI." Can we remove it and just leave it at "The CLI can be used to run, configure, and test GeoIPS. It can also print lists or descriptions of available data or functionality."

jsolbrig commented on docs/source/userguide/command_line.rst:
A few general comments on organization:

- I would like to move all of the discussion of aliases out of the individual sections and into a single section that shows how to discover the available aliases. I don't think we need to explicitly document each alias here.
- Maybe this is better in alphabetical order like it currently is, but I feel like ``list`` might be better above ``describe`` because I expect people will want to get a list of artifacts then, using that list, get descriptions of individual artifacts.
- There are parts of this that seem difficult to maintain. Would it make sense to auto-generate more portions of this?

evrose54 replied:
I agree that reorganizing the documentation here is a good idea. It makes logical sense for list to be first, for the reasons Jeremy mentioned. As for auto-generating this documentation, I'm not sure how easy that would be. We can definitely look into it, however if we plan to keep what's in the CLI as-is, it might just be worth leaving the documentation hardcoded and expanding the documentation as required by future updates.

jsolbrig commented on docs/source/userguide/command_line.rst:
```diff
@@ -1,508 +1,520 @@
-:ref:`retrieve information<information_retrieval>` from GeoIPS and CLI commands which
-:ref:`perform processes<performing_processes>` via GeoIPS. We'll dive into each of those
-sections now.
+ - `Information retrieval`_ commands
```
I'm not sure that I like "Information retrieval commands" or "Action performing commands". Ideas:

### "Information retrieval commands" alternatives
"Passive commands"
"Discovery commands"
"Introspection commands"
"Introspective commands"

### "Action performing commands" alternatives
"Active commands"
"Action commands"
"Operational commands"
"Execution commands"

biosafetylvl5 replied:
I like "Discovery Commands" and "Action commands"

jsolbrig commented on docs/source/userguide/command_line.rst:
```diff
@@ -1,508 +1,520 @@
-artifacts based on the sub-commands provided. These two top-level commands are
-``describe`` (``desc``) and ``list`` (``ls``). Each command shown below can be ran with
-a ``-h`` flag to provide help instructions for the command you are running.
+The CLI Implements two top-level commands which retrieve information about GeoIPS
```
commands **that** retrieve information (I think?)

jsolbrig commented on docs/source/userguide/command_line.rst:
```diff
@@ -1,508 +1,520 @@
+artifacts: ``describe`` and ``list`` (``ls``).
 
-.. _geoips_describe:
+Each command shown below can be ran with a ``-h``
+flag to provide associated help instructions.
```
Maybe move this to the top section and expand slightly? For example:

To retrieve information about the CLI's commands and arguments, the ``-h/--help`` argument can be provided at any time. For example, each of the following will return different, context dependent, help information:
- ``geoips -h``
- ``geoips list -h``
- ``geoips list algorithms -h``
- ``geoips run -h``

jsolbrig commented on docs/source/userguide/command_line.rst:
```diff
@@ -1,508 +1,520 @@
 
-Implemented Mode Options
-::
+    geoips list interfaces --package_name geoips
+    geoips list <interface_name> -p <package_name>
```
Maybe this would be more clear if the example used a package other than ``geoips`` like ``geoips_clavrx``.

evrose54 replied:
Agreed.

jsolbrig commented on docs/source/userguide/command_line.rst:
```diff
@@ -1,508 +1,520 @@
-you want to use these commands in shorthand style. One of the commands below lists
-plugins from a certain GeoIPS package.
-::
+``list plugins`` is a sub-command of list.
```
This is stated in quite a few places. I don't think we need to state "Y is a subcommand of X". That should be obvious from the documentation flow I think.

evrose54 replied:
Fair point.

jsolbrig reviewed (changes requested):
Here is my initial review. I think I'll want to look at this again when the current comments are addressed.

evrose54 commented on docs/source/userguide/command_line.rst:
```diff
@@ -1,508 +1,520 @@
-``describe`` (``desc``) and ``list`` (``ls``). Each command shown below can be ran with
-a ``-h`` flag to provide help instructions for the command you are running.
+The CLI Implements two top-level commands which retrieve information about GeoIPS
+artifacts: ``describe`` and ``list`` (``ls``).
```
If we provide an alias of ``ls`` here should we also provide ``desc``?

evrose54 commented on docs/source/userguide/command_line.rst:
```diff
@@ -1,508 +1,520 @@
+``list <interface_name>`` is a sub-command of list.
 
-.. _geoips_list_interfaces:
+It returns a list of an interfaces plugins with the following plugin information:
```
interface's

evrose54 commented on docs/source/userguide/command_line.rst:
```diff
@@ -1,508 +1,520 @@
-unit tests from a certain GeoIPS package.
-::
+.. note::
+    For this command to find your unit tets, you must
```
tests

evrose54 reviewed (changes requested):
After going through this it's clear to me that I'm over-verbose in my writing style. Documentation should be short and to the point and I think this update captures that! @biosafetylvl5 I left a few comments that need to be addressed, and it looks like Jeremy left quite a few that I agree with as well. Once those have been completed, I'm good to see this merged! Thanks for tackling this!

biosafetylvl5 commented:
I've addressed all the comments. Changes:

- Moved list above describe 
- Consolidated aliases into a single section rather than repeating them throughout
- Adopted "Discovery Commands" and "Action Commands"
- grammar, typos and clarity
- Fixed the errors we caught
- Removed "X is a subcommand of Y" statements

evrose54 commented on docs/source/concepts/functionality/command-line/index.rst:
```diff
@@ -553,126 +496,117 @@ New CLI-based Procflow (abi.static.Infrared.imagery_annotated.sh)
 
-::
+``sector`` produces a .png image based on the provided sector plugin name. The sector
+must be an entry within any Plugin Package's registered_plugins.(yaml/json) file.
```
The sector actually just has to be an entry in the .json file. That's what we use under the hood. .yaml is just for viewing.

evrose54 reviewed (dismissed):
Looks good to me. Thanks for doing this! Just an idea, you don't have to implement it if you disagree. Would it be useful to provide example output for some of the example commands that you provided? I.e. the output table from a list command, the output of a describe command, etc. I added one other comment, but besides that it looks good! I'll update this documentation once merged as there is another PR (#974) that adds some new functionality.

biosafetylvl5 commented:
Thanks! :slightly_smiling_face: I think including CLI output would be helpful, but I also don't want to have to update it every time we update the CLI... probably a good case for some auto-doc stuff. Feel free to add to this PR / open an issue for it if you think it's something we definitely should add!

mindyls commented on docs/source/concepts/functionality/command-line/command_line_autodoc.rst:
```diff
@@ -0,0 +1,12 @@
+
+.. dropdown:: Distribution Statement
+
+ This source code is protected under the license referenced at
+ https://github.com/NRLMMD-GEOIPS.
```
I think the latest verbiage for this is "subject to" not "protected under"

biosafetylvl5 replied:
Fixed

mindyls commented on docs/source/concepts/functionality/command-line/index.rst:
```diff
@@ -1,533 +1,478 @@
 
- | # # # This source code is subject to the license referenced at
- | # # # https://github.com/NRLMMD-GEOIPS.
+ This source code is protected under the license referenced at
+ https://github.com/NRLMMD-GEOIPS.
```
This source code is subject to the license referenced at
https://github.com/NRLMMD-GEOIPS.

biosafetylvl5 replied:
Fixed
